lib/rubocop/cop/style/empty_case_condition.rb

```diff
@@ -70,21 +70,24 @@ def correct_case_whens(corrector, case_node, when_nodes)
         end
 
         # Since an if condition containing commas is not syntactically valid, we
-        # correct `when x,y` to `if [x,y].any?`.
+        # correct `when x, y` to `if x || y`.
         def correct_multiple_alternative_whens(corrector, when_nodes)
           when_nodes.each do |when_node|
-            children = when_node.children
+            *children, _ = when_node.children
 
             # In `when a; r` we have two children: [a, r].
             # In `when a, b, c; r` we have 4.
-            next unless children.size > 2
+            # In `when a, b then r` we have 3.
```

Comment:
I moved the comment up. I agree, it was a bit confusing. When a condition uses `then` it works very similarly to the case with the semicolon. The last child is the body of the `when` node. 

Right now all tests are passing so the logic is working. I would be happy to change the line, but I would like to have a test that proves that the line needs to be changed.